src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -1,100 +1,61 @@
 package com.rybka.service;
 
-import com.google.gson.JsonElement;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
-import com.rybka.dao.HibernateDAO;
-import com.rybka.model.Currency;
-
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.HttpURLConnection;
-import java.net.URL;
-
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.exception.NullResponseException;
+import com.rybka.exception.ZeroCurrencyRateException;
+import com.rybka.model.ConvertedCurrency;
+import com.rybka.model.ExchangeRate;
+import lombok.Data;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+@Data
 public class ExchangeService {
 
-    private static final String API_KEY = "4e0bbdc44fcdf05612fa0882";
-
-    private Double currencyValue;
-    private String baseCurrencyAbbreviation;
-    private String targetCurrencyAbbreviation;
-    private Double amount;
-    private Double total;
-
-    public void loadCurrencyOf(String userBaseCurrencyAbbreviation, String userTargetCurrencyAbbreviation) {
-        String url_str = String.format("https://prime.exchangerate-api.com/v5/%s/latest/%s",
-                API_KEY, userBaseCurrencyAbbreviation);
-
-        try {
-            URL url = new URL(url_str);
-            HttpURLConnection request = (HttpURLConnection) url.openConnection();
-            request.connect();
-
-            JsonParser jp = new JsonParser();
-            JsonElement root = jp.parse(new InputStreamReader((InputStream) request.getContent()));
-            JsonObject jsonobj = root.getAsJsonObject();
-
-            JsonObject jsonObject = jsonobj.get("conversion_rates").getAsJsonObject();
 
-            currencyValue = jsonObject.get(userTargetCurrencyAbbreviation).getAsDouble();
-            baseCurrencyAbbreviation = userBaseCurrencyAbbreviation;
-            targetCurrencyAbbreviation = userTargetCurrencyAbbreviation;
-        } catch (Exception e) {
-            System.out.println("Incorrect exchange data!");
-            System.exit(3);
+    public ExchangeRate loadCurrencyOf(String userBaseCurrency, String userTargetCurrency) throws NullResponseException, ZeroCurrencyRateException {
```

Comment:
I'd retrieve a list of ExhangeRate for baseCurrency here. 

List<ExchangeRate> listRates = loadCurrency(baseCurrency);

Reply:
fixed